### PR TITLE
PR for review of the milestone 3

### PR DESCRIPTION
**Pull request by kalavhan**
Devise behavior modified to accomplish the desired result :
- Signup and login in the same pages as Facebook has it
     + This required to modify the actions in the devise controller
     + The login form is a partial that we render in the header conditionally if no user has signed in
     + The user is redirected to Signup page if he is not signed in

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
```
Comment (samgaco) [inline app/controllers/posts_controller.rb:3]: - [x] There should be also a controller for the user displaying all the users, with a page to show a list of all available users. [Requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone3/userctr.png).

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authorize, only: [:index]
+
+  def index
+    @post = Post.new
+    @posts = Post.where(user_id: current_user.id)
```
Comment (samgaco) [inline app/controllers/posts_controller.rb]: - [x] SUGGESTION Have into account that everything connected with data processing should be encapsulated in models and used in controllers.

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @kalavhan and @fegzycole :high_brightness: 

Everything looking right! just one thing to correct. We should have a users controller and a index of all the users.

Work on that and submit back for another review. Good spirit! :muscle: 
--------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Review by Ispirett — APPROVED** [review]
## Status Approved :+1: :100: 
Great job @kalavhan  @fegzycole  you have done well move  along. :+1:

**Pull request merged**